pkgs/pkgs-lib/tests/formats.nix

```diff
@@ -147,6 +147,51 @@ in runBuildTests {
     '';
   };
 
+  testKeyValueAtoms = {
+    drv = evalFormat formats.keyValue {} {
+      bool = true;
+      int = 10;
+      float = 3.141;
+      str = "string";
+    };
+    expected = ''
+      bool=true
+      float=3.141000
+      int=10
+      str=string
+    '';
+  };
+
+  testKeyValueDuplicateKeys = {
+    drv = evalFormat formats.keyValue { listsAsDuplicateKeys = true; } {
+      bar = [ null true "test" 1.2 10 ];
+      baz = false;
+      qux = "qux";
+    };
+    expected = ''
+      bar=null
```

Comment:
I suppose rendering `null` as `"null"` is consistent with the other formats.
This will be a small UX problem when people expect something specific from `null`, but they get `"null"` instead because nulls aren't a thing in some config format. It'd be easier just to leave it out for now. If you need it, you can make it configurable with the format parameters.

Reply:
> I suppose rendering `null` as `"null"` is consistent with the other formats. 
:+1: 

While working on another module, I realized that that it may be good idea to have an option like `ignoreNullLines` later on.

Reply:
What would that do? Would it not print the key/value if value == null?

Reply:
Yes

Reply:
> What would that do? Would it not print the key/value if value == null?

That's one way to do it, matching `mkDerivation`. Another way is to remove it from the type so it isn't advertised and won't be used.

Reply:
It shouldn't be removed or it won't be possible to format 'types.nullOr' keys.

Reply:
Modules that rely on that would not configure the format to disallow `null`.

A dynamic set of keys can also be achieved with `optionalAttrs`, `filterAttrs` and as is the case with `attrsOf`, `mkIf`.

It's ok to add features later, so I consider this thread resolved.